syntax = "proto2";

package mlflow;

import "scalapb/scalapb.proto";

option java_package = "org.mlflow.api.proto";
option py_generic_services = true;
option (scalapb.options) = {flat_package: true};

// Generic status enum for MLflow jobs.
// Can be used across different job types (optimization, evaluation, etc.).
enum JobStatus {
  JOB_STATUS_UNSPECIFIED = 0;

  // Job is queued, waiting to start.
  JOB_STATUS_PENDING = 1;

  // Job is currently running.
  JOB_STATUS_IN_PROGRESS = 2;

  // Job completed successfully.
  JOB_STATUS_COMPLETED = 3;

  // Job failed with an error.
  JOB_STATUS_FAILED = 4;

  // Job was canceled by user.
  JOB_STATUS_CANCELED = 5;
}

// Generic job state message combining status with metadata.
// Provides a unified way to represent job state across different job types.
message JobState {
  // Current status of the job.
  optional JobStatus status = 1;

  // Error message if the job failed.
  // Only set when status is JOB_STATUS_FAILED.
  optional string error_message = 2;

  // Additional metadata as key-value pairs.
  // Can be used to store job-specific state information.
  map<string, string> metadata = 3;
}
